sumo/src/main.rs

```diff
@@ -0,0 +1,193 @@
+            elevation: Distance::ZERO,
+            intersection_type: IntersectionType::StopSign,
+            orig_id: osm::NodeID(123),
+            incoming_lanes: Vec::new(),
+            outgoing_lanes: Vec::new(),
+            roads: BTreeSet::new(),
+        });
+        ids_intersections.insert(junction.id, id);
+    }
+    let mut roads: Vec<Road> = Vec::new();
+    let mut lanes = Vec::new();
+    let mut ids_lanes: BTreeMap<sumo::LaneID, LaneID> = BTreeMap::new();
+    for (_, edge) in network.normal_edges {
+        let src_i = ids_intersections[&edge.from];
+        let dst_i = ids_intersections[&edge.to];
+        // SUMO has one edge in each direction, but ABST has bidirectional roads. Detect if this
+        // edge is the reverse of one we've already handled.
+        let (road_id, direction) =
+            if let Some(r) = roads.iter().find(|r| r.dst_i == src_i && r.src_i == dst_i) {
+                (r.id, map_model::Direction::Back)
+            } else {
+                (RoadID(roads.len()), map_model::Direction::Fwd)
+            };
+
+        let mut lanes_rtl: Vec<(LaneID, map_model::Direction, LaneType)> = Vec::new();
+        for lane in &edge.lanes {
+            let lane_id = LaneID(lanes.len());
+            ids_lanes.insert(lane.id.clone(), lane_id);
+            let lane_type = if lane.allow == vec!["pedestrian".to_string()] {
+                LaneType::Sidewalk
+            } else if lane.allow == vec!["bicycle".to_string()] {
+                LaneType::Biking
+            } else if lane.allow == vec!["rail_urban".to_string()] {
+                LaneType::LightRail
+            } else {
+                LaneType::Driving
+            };
+            lanes.push(Lane {
+                id: lane_id,
+                parent: road_id,
+                lane_type,
+                lane_center_pts: lane.center_line.clone(),
+                width: lane.width,
+
+                src_i,
+                dst_i,
+
+                bus_stops: BTreeSet::new(),
+
+                driving_blackhole: false,
+                biking_blackhole: false,
+            });
+            // These seem to appear in the XML from right to left
+            lanes_rtl.push((lane_id, direction, lane_type));
+        }
+
+        if direction == map_model::Direction::Fwd {
+            // Make a new road
+            intersections[src_i.0].roads.insert(road_id);
+            intersections[dst_i.0].roads.insert(road_id);
+            let speed_limit = edge.lanes[0].speed;
+
+            let mut tags = BTreeMap::new();
+            tags.insert("id".to_string(), edge.id.0.clone());
+            if let Some(name) = &edge.name {
+                tags.insert("name".to_string(), name.clone());
+            }
+            let parts: Vec<&str> = edge.edge_type.split(".").collect();
+            // "highway.footway"
+            if parts.len() == 2 {
+                tags.insert(parts[0].to_string(), parts[1].to_string());
+            }
```

Comment:
Do we encoutner situations where `parts.len() != 2`? Is it ok to just ignore it?

Reply:
Haven't seen that happen yet, so added an error if it does pop up